core/kernel/dt_driver_test.c

```diff
@@ -523,3 +569,142 @@ static TEE_Result dt_test_rstctrl_provider_probe(const void *fdt, int offs,
 RSTCTRL_DT_DECLARE(dt_test_rstctrl_provider, "linaro,dt-test-provider",
 		   dt_test_rstctrl_provider_probe);
 #endif /* CFG_DRIVERS_RSTCTRL */
+
+#ifdef CFG_DRIVERS_GPIO
+#define DT_TEST_GPIO_COUNT	2
+
+#define DT_TEST_GPIO0_PIN	1
+#define DT_TEST_GPIO0_FLAGS	GPIO_ACTIVE_LOW
+#define DT_TEST_GPIO1_PIN	2
+#define DT_TEST_GPIO1_FLAGS	GPIO_PULL_UP
+
+struct dt_test_gpio {
+	unsigned int pin;
+	unsigned int flags;
+	struct gpio_chip gpio_chip;
+};
+
+static struct dt_test_gpio *to_test_gpio(struct gpio_chip *chip)
+{
+	return container_of(chip, struct dt_test_gpio, gpio_chip);
+}
+
+static enum gpio_dir dt_test_gpio_get_direction(struct gpio_chip *chip,
+						unsigned int gpio_pin)
+{
+	struct dt_test_gpio *dtg = to_test_gpio(chip);
+
+	if (dtg->pin != gpio_pin)
+		panic("Invalid gpio number");
+
+	return GPIO_DIR_IN;
+}
+
+static void dt_test_gpio_set_direction(struct gpio_chip *chip,
+				       unsigned int gpio_pin,
+				       enum gpio_dir direction __unused)
+{
+	struct dt_test_gpio *dtg = to_test_gpio(chip);
+
+	if (dtg->pin != gpio_pin)
+		panic("Invalid gpio number");
+}
+
+static enum gpio_level dt_test_gpio_get_value(struct gpio_chip *chip,
+					      unsigned int gpio_pin)
+{
+	struct dt_test_gpio *dtg = to_test_gpio(chip);
+
+	if (dtg->pin != gpio_pin)
+		panic("Invalid gpio number");
+
+	return GPIO_LEVEL_HIGH;
+}
+
+static void dt_test_gpio_set_value(struct gpio_chip *chip,
+				   unsigned int gpio_pin,
+				   enum gpio_level value __unused)
+{
+	struct dt_test_gpio *dtg = to_test_gpio(chip);
+
+	if (dtg->pin != gpio_pin)
+		panic("Invalid gpio number");
+}
+
+static const struct gpio_ops dt_test_gpio_ops = {
+	.get_direction = dt_test_gpio_get_direction,
+	.set_direction = dt_test_gpio_set_direction,
+	.get_value = dt_test_gpio_get_value,
+	.set_value = dt_test_gpio_set_value,
+};
+
+static struct gpio *dt_test_gpio_get_dt(struct dt_driver_phandle_args *a,
+					void *data, TEE_Result *res)
+{
+	struct gpio *gpio = NULL;
+	struct dt_test_gpio *gpios = (struct dt_test_gpio *)data;
+
+	gpio = gpio_get_dt_two_cells(a, res);
+	if (*res)
+		return NULL;
+
+	switch (gpio->pin) {
+	case DT_TEST_GPIO0_PIN:
+		gpio->chip = &gpios[0].gpio_chip;
+		if (gpio->dt_flags != gpios[0].flags) {
+			EMSG("Unexpected dt_flags %u", gpio->dt_flags);
```

Comment:
`dt_flags` field is `uint32_t` hence must change from `"%u"` to `"%"PRIu32`.
By the way, as it's a bitmask value, I think an hexadecimal print would be more applicable:
`EMSG("Unexpected dt_flags %#"PRIx32, gpio->dt_flags);`

Ditto at line 663.


Note another opion could be to use `unsigned int` type from `dt_flags` field in _gpio.h_ (which I think would make sense).